openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +388,226 @@ def debug_cohort_mgmt(request, course_key_string):
+        """
+        Endpoint to get either one or all cohorts.
+        """
+        course_key, course = api.get_course(request, course_key_string)
+        if not cohort_id:
+            all_cohorts = cohorts.get_course_cohorts(course)
+            paginator = NamespacedPageNumberPagination()
+            paginator.max_page_size = 100
+            page = paginator.paginate_queryset(all_cohorts, request)
+            response = [_get_cohort_representation(c, course) for c in page]
+            return Response(response, status=status.HTTP_200_OK)
+        else:
+            cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
+            return Response(_get_cohort_representation(cohort, course), status=status.HTTP_200_OK)
+
+    def post(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to create a new cohort, must not include cohort_id.
+        """
+        if cohort_id is not None:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Please user the parent endpoint.',
+                                 'wrong-endpoint')
+        course_key, course = api.get_course(request, course_key_string)
+        name = request.data.get('name')
+        if not name:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Cohort name must be specified.',
+                                 'missing-cohort-name')
+        assignment_type = request.data.get('assignment_type')
+        if not assignment_type:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Assignment type must be specified.',
+                                 'missing-assignment-type')
+        return Response(_get_cohort_representation(cohorts.add_cohort(course_key, name, assignment_type), course))
+
+    def patch(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to update a cohort information, including:
+        """
+        name = request.data.get('name')
+        assignment_type = request.data.get('assignment_type')
+        if None in {cohort_id, name, assignment_type}:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Request must include cohort id, name and assignment type.',
+                                 'missing-field')
+        course_key, _ = api.get_course(request, course_key_string)
+        cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
+        if name != cohort.name:
+            if cohorts.is_cohort_exists(course_key, name):
+                raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                     'A cohort with the same name already exists.',
+                                     'cohort-name-exists')
+            cohort.name = name
+            cohort.save()
+        cohorts.set_assignment_type(cohort, assignment_type)
```

Comment:
Maybe catch the ValueError here and raise and appropriate api_error.

Reply:
👍